src/dune_file.mli

```diff
@@ -113,6 +113,7 @@ module Buildable : sig
     { loc                      : Loc.t
     ; modules                  : Ordered_set_lang.t
     ; modules_without_implementation : Ordered_set_lang.t
+    ; variants                 : Variant.Set.t
```

Comment:
I'm not so sure it should be allowed to specify `variants` for libraries. At least not the way it's implemented in this PR. Am I correct to note that this information is omitted from the `dune-package` file? If that's the case, I'm thinking this could lead to some surprising differences for internal and external libraries.

Reply:
This is actually a mistake. This should probably live in `Executable.t`, right ?

Reply:
Yeah, indeed that was my original problem. There's probably a way to make it work with libraries as well but let's not complicate things further.